hack/lib/common.sh

```diff
@@ -474,28 +515,96 @@ function kube::common::start_controller_manager {
       cloud_config_arg+=("--external-cloud-volume-plugin=${CLOUD_PROVIDER}")
       cloud_config_arg+=("--cloud-config=${CLOUD_CONFIG}")
     fi
+
     CTLRMGR_LOG=${LOG_DIR}/kube-controller-manager.log
-    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" kube-controller-manager \
-      --v="${LOG_LEVEL}" \
-      --allocate-node-cidrs="${KUBE_CONTROLLER_MANAGER_ALLOCATE_NODE_CIDR}" \
-      --cluster-cidr="${KUBE_CONTROLLER_MANAGER_CLUSTER_CIDR}" \
-      --vmodule="${LOG_SPEC}" \
-      --service-account-private-key-file="${SERVICE_ACCOUNT_KEY}" \
-      --root-ca-file="${ROOT_CA_FILE}" \
-      --cluster-signing-cert-file="${CLUSTER_SIGNING_CERT_FILE}" \
-      --cluster-signing-key-file="${CLUSTER_SIGNING_KEY_FILE}" \
-      --enable-hostpath-provisioner="${ENABLE_HOSTPATH_PROVISIONER}" \
-      ${node_cidr_args[@]+"${node_cidr_args[@]}"} \
-      --pvclaimbinder-sync-period="${CLAIM_BINDER_SYNC_PERIOD}" \
-      --feature-gates="${FEATURE_GATES}" \
-      "${cloud_config_arg[@]}" \
-      --kubeconfig "${kubeconfigfilepaths}" \
-      --use-service-account-credentials \
-      --controllers="${KUBE_CONTROLLERS}" \
-      --leader-elect=false \
-      --cert-dir="${CERT_DIR}" \
-      --default-network-template-path="${ARKTOS_NETWORK_TEMPLATE}" \
-      --master="https://${API_HOST}:${API_SECURE_PORT}" >"${CTLRMGR_LOG}" 2>&1 &
+
```

Comment:
the construction of the parameters has a lot dup code which can be removed by having a base parameter list and then add more as needed for different cases. should have a TODO here for future work tracking.

Reply:
We can add a TODO here. But just as a heads up, I tried reuse base parameters before but it did not work.

Reply:
https://github.com/CentaurusInfra/arktos/issues/1098